Sources/Ink/Internal/CodeBlock.swift

```diff
@@ -13,16 +13,20 @@ internal struct CodeBlock: Fragment {
     private var code: String
 
     static func read(using reader: inout Reader) throws -> CodeBlock {
-        try require(reader.readCount(of: marker) == 3)
+        let startingMarkerCount = reader.readCount(of: marker)
+        try require(startingMarkerCount >= 3)
 
-        let language = reader.readUntilEndOfLine()
+        var language = reader.readUntilEndOfLine().drop(while: {char in char.isWhitespace})
+        while let l = language.last, l.isWhitespace {
+            language = language.dropLast()
+        }
         var code = ""
 
         while !reader.didReachEnd {
             if code.last == "\n", reader.currentCharacter == marker {
                 let markerCount = reader.readCount(of: marker)
 
-                if markerCount == 3 {
+                if markerCount == startingMarkerCount {
```

Comment:
The [CommonMark spec](https://spec.commonmark.org/0.29/#example-94) allows for this to be more than the starting marker count. Might be worth addressing that while you're working on this.

Reply:
I am writing a separate project that is building a set of swift test cases from the CommonMark spec that can be used as a TDD source for all these cases; or at least a definitive compliance test suite.  

I bumped into many other things that I left out of the pull request to keep it simple.

Reply:
Sounds good

Reply:
Agree that introducing multiple changes piece by piece is a great plan. Also makes it much more likely that I'll be able to do a quick review on each change 🙂